src/pydex_app/orderbook.py

```diff
@@ -117,6 +228,73 @@ def get_asks(
             asks = sorted(asks, key=lambda o: o.sort_price, reverse=True)
         return paginate(asks, page=page, per_page=per_page), asks_count
 
+    @classmethod
+    def get_orders(  # pylint: disable=too-many-locals
+        cls,
+        maker_asset_proxy_id=None,
+        taker_asset_proxy_id=None,
+        maker_asset_address=None,
+        taker_asset_address=None,
+        exchange_address=None,
+        sender_address=None,
+        maker_asset_data=None,
+        taker_asset_data=None,
+        maker_address=None,
+        taker_address=None,
+        fee_recipient_address=None,
+        page=DEFAULT_PAGE,
+        per_page=DEFAULT_PER_PAGE
+    ):
+        """Retrieves a list of orders given query parameters.
+
+        Keyword arguments:
+        maker_asset_proxy_id -- string proxy id for the maker asset
+        taker_asset_proxy_id -- string proxy id for the taker asset
+        maker_asset_address -- string address for maker asset
+        taker_asset_address -- string address for taker asset
+        exchange_address -- string address for the 0x exchange contract
+        sender_address -- string address for the party reponsible to broadcast the order
+        maker_asset_data -- string asset_data for maker side (i.e. `maker_asset_data`)
+        taker_asset_data -- string asset_data for taker side (i.e. `taker_asset_data`)
+        maker_address -- string address for the maker
+        maker_address -- string address for the taker
+        fee_recipient_address -- string address for the fee recipient
+        """
+        pre_filter = dict(
+            exchange_address=normalize_query_param(exchange_address),
+            sender_address=normalize_query_param(sender_address),
+            maker_asset_data=normalize_query_param(maker_asset_data)
+            or (adu.encode_erc20_asset_data(maker_asset_address) if maker_asset_address else None),
+            taker_asset_data=normalize_query_param(taker_asset_data)
+            or (adu.encode_erc20_asset_data(taker_asset_address) if taker_asset_address else None),
+            maker_address=normalize_query_param(maker_address),
+            taker_address=normalize_query_param(taker_address),
+            fee_recipient_address=normalize_query_param(fee_recipient_address)
+        )
+        filter_object = {k: v for k, v in pre_filter.items() if v is not None}
+        normalized_maker_asset_proxy_id = normalize_query_param(maker_asset_proxy_id)
+        normalized_taker_asset_proxy_id = normalize_query_param(taker_asset_proxy_id)
+        if maker_asset_proxy_id and taker_asset_proxy_id:
```

Comment:
check my note above about how to refactor this to be less code and easier to read.